### PR TITLE
[action] Support Reset next snapshot id of consumer id.

### LINKED ISSUE #1574: [Feature] Support Reset next snapshot id of consumer id
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Currently paimon does NOT support reset next snapshot id of consumer id which is not convenient in `DATA FIXING` scenario

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Linked issue: close #1574 

### Tests

FlinkActionsE2eTest#testRecordConsumer
ConsumerActionITCase#testRecordConsumer

### CODE REVIEW

File: docs/content/maintenance/manage-consumers.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+```bash
+<FLINK_HOME>/bin/flink run \
+    /path/to/paimon-flink-action-{{< version >}}.jar \
+    record-consumer \
+    --warehouse <warehouse-path> \
+    --database <database-name> \ 
+    --table <table-name> \
+    --consumer-id <consumer-id> \
+    --snapshot <next-snapshot-id> \
```
Comment (yuzelin) [inline docs/content/maintenance/manage-consumers.md:51]: How about renaming this argument to `--next-snapshot`?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/consumer/RecordConsumerAction.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+package org.apache.paimon.flink.action.consumer;
+
+import org.apache.paimon.flink.action.TableActionBase;
+
+import java.util.Map;
+
+/** Record consumer action for Flink. */
+public class RecordConsumerAction extends TableActionBase {
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/consumer/RecordConsumerAction.java:26]: I think it's unnecessary to place this action to a new package folder.

File: paimon-e2e-tests/src/test/java/org/apache/paimon/tests/FlinkActionsE2eTest.java [hunk 3]
```diff
@@ -400,6 +400,79 @@ public void testCreateAndDeleteTag() throws Exception {
...
+                catalogDdl,
+                useCatalogCmd,
+                tableDdl,
+                createResultSink("_consumers2", "consumer_id STRING, next_snapshot_id BIGINT"));
+
+        // The value of next_snapshot_id in the result is 4 instead of 1,
+        // because the streaming read task is running, and the consumer `myid` is reset again,
+        // overwriting the value written by the action job.
+        checkResult("myid, 4");
+    }
```
Comment (yuzelin) [inline paimon-e2e-tests/src/test/java/org/apache/paimon/tests/FlinkActionsE2eTest.java:474]: I think we can remove this test.

**Review by yuzelin — COMMENTED** [review]
Thanks for contributing~ left some comments.

**Comment by zhuangchong** [comment]
> Thanks for contributing~ left some comments.

@yuzelin Thanks. done.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/TableActionBase.java [hunk 4]
```diff
@@ -57,7 +57,7 @@ public abstract class TableActionBase extends ActionBase {
 
     protected Table table;
 
-    TableActionBase(
+    protected TableActionBase(
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/TableActionBase.java:60]: This can be reset.

**Review by yuzelin — COMMENTED** [review]
A minor comment.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/TableActionBase.java (hunk 4 above)
Comment (zhuangchong) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/TableActionBase.java]: Thanks.

**Comment by yuzelin** [comment]
Not sure that whether it's necessary to introduce an method in public `Table` interface.
@JingsongLi Please take a look.

File: docs/content/maintenance/manage-consumers.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Manage Consumers
+
+## Record Consumer
```
Comment (JingsongLi) [inline docs/content/maintenance/manage-consumers.md:29]: Just document this with consumer. in query-table.

File: docs/content/maintenance/manage-consumers.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+You can add or reset a consumer with a given consumer ID and next snapshot ID.
+
+{{< hint info >}}
+First, you need to stop the streaming task using this consumer ID, and then execute the record consumer action job.
+{{< /hint >}}
+
+{{< tabs "record-consumer" >}}
```
Comment (JingsongLi) [inline docs/content/maintenance/manage-consumers.md:37]: Is there a better name?

File: paimon-core/src/main/java/org/apache/paimon/table/Table.java [hunk 7]
```diff
@@ -79,6 +79,9 @@ public interface Table extends Serializable {
     @Experimental
     void rollbackTo(String tagName);
 
+    /** Record the consumer of the table through consumer id and next snapshot id. */
+    void recordConsumer(String consumerId, long nextSnapshotId);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/Table.java:83]: Don't introduce public API now. It is not so stable.

File: docs/content/maintenance/manage-consumers.md (hunk 5 above)
Comment (zhuangchong) [reply docs/content/maintenance/manage-consumers.md]: done.

File: docs/content/maintenance/manage-consumers.md (hunk 6 above)
Comment (zhuangchong) [reply docs/content/maintenance/manage-consumers.md]: Has been modified to `reset-consumer`.

File: paimon-core/src/main/java/org/apache/paimon/table/Table.java (hunk 7 above)
Comment (zhuangchong) [reply paimon-core/src/main/java/org/apache/paimon/table/Table.java]: done.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ResetConsumerAction.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        this.consumerId = consumerId;
+        this.nextSnapshotId = nextSnapshotId;
+    }
+
+    @Override
+    public void run() throws Exception {
+        FileStoreTable dataTable = (FileStoreTable) table;
+        ConsumerManager consumerManager =
+                new ConsumerManager(dataTable.fileIO(), dataTable.location());
+        consumerManager.recordConsumer(consumerId, new Consumer(nextSnapshotId));
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ResetConsumerAction.java:50]: resetConsumer

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ResetConsumerActionFactory.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                        catalogConfig,
+                        consumerId,
+                        nextSnapshotId);
+        return Optional.of(action);
+    }
+
+    @Override
+    public void printHelp() {
+        System.out.println(
+                "Action \"reset-consumer\" record a consumer from the given next snapshot.");
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ResetConsumerActionFactory.java:61]: reset

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ResetConsumerActionFactory.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        System.out.println();
+
+        System.out.println("Syntax:");
+        System.out.println(
+                "  record-consumer --warehouse <warehouse-path> --database <database-name> "
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ResetConsumerActionFactory.java:66]: reset

**Review by JingsongLi — APPROVED** [review]
+1

**Comment by zhuangchong** [comment]
> +1

Thanks. @JingsongLi

**Pull request merged**